gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergPartitionDatasetTest.java

```diff
@@ -0,0 +1,323 @@
+  private TableMetadata srcTableMetadata;
+  private TableMetadata destTableMetadata;
+  private FileSystem sourceFs;
+  private FileSystem targetFs;
+  private IcebergPartitionDataset icebergPartitionDataset;
+  private MockedStatic<IcebergPartitionFilterPredicateUtil> icebergPartitionFilterPredicateUtil;
+  private static final String SRC_TEST_DB = "srcTestDB";
+  private static final String SRC_TEST_TABLE = "srcTestTable";
+  private static final String SRC_WRITE_LOCATION = SRC_TEST_DB + "/" + SRC_TEST_TABLE + "/data";
+  private static final String DEST_TEST_DB = "destTestDB";
+  private static final String DEST_TEST_TABLE = "destTestTable";
+  private static final String DEST_WRITE_LOCATION = DEST_TEST_DB + "/" + DEST_TEST_TABLE + "/data";
+  private static final String TEST_ICEBERG_PARTITION_COLUMN_NAME = "testPartition";
+  private static final String TEST_ICEBERG_PARTITION_COLUMN_VALUE = "testValue";
+  private final Properties copyConfigProperties = new Properties();
+  private final Properties properties = new Properties();
+  private List<String> srcFilePaths;
+
+  private static final URI SRC_FS_URI;
+  private static final URI DEST_FS_URI;
+
+  static {
+    try {
+      SRC_FS_URI = new URI("abc", "the.source.org", "/", null);
+      DEST_FS_URI = new URI("xyz", "the.dest.org", "/", null);
+    } catch (URISyntaxException e) {
+      throw new RuntimeException("should not occur!", e);
+    }
+  }
+
+  @BeforeMethod
+  public void setUp() throws Exception {
+    setupSrcFileSystem();
+    setupDestFileSystem();
+
+    TableIdentifier tableIdentifier = TableIdentifier.of(SRC_TEST_DB, SRC_TEST_TABLE);
+
+    srcIcebergTable = Mockito.mock(IcebergTable.class);
+    destIcebergTable = Mockito.mock(IcebergTable.class);
+
+    srcTableMetadata = Mockito.mock(TableMetadata.class);
+    destTableMetadata = Mockito.mock(TableMetadata.class);
+    Mockito.when(destTableMetadata.spec()).thenReturn(Mockito.mock(PartitionSpec.class));
+
+    Mockito.when(srcIcebergTable.getTableId()).thenReturn(tableIdentifier);
+    Mockito.when(destIcebergTable.getTableId()).thenReturn(tableIdentifier);
+    Mockito.when(srcIcebergTable.accessTableMetadata()).thenReturn(srcTableMetadata);
+    Mockito.when(destIcebergTable.accessTableMetadata()).thenReturn(destTableMetadata);
+    Mockito.when(srcIcebergTable.getDatasetDescriptor(Mockito.any())).thenReturn(Mockito.mock(DatasetDescriptor.class));
+    Mockito.when(destIcebergTable.getDatasetDescriptor(Mockito.any())).thenReturn(Mockito.mock(DatasetDescriptor.class));
+
+    icebergPartitionFilterPredicateUtil = Mockito.mockStatic(IcebergPartitionFilterPredicateUtil.class);
+    icebergPartitionFilterPredicateUtil
+        .when(() -> IcebergPartitionFilterPredicateUtil.getPartitionColumnIndex(Mockito.anyString(), Mockito.any(), Mockito.any()))
+        .thenReturn(Optional.of(0));
+
+    copyConfigProperties.setProperty("data.publisher.final.dir", "/test");
+    srcFilePaths = new ArrayList<>();
```

Comment:
confusing that you already have `.clear()` in your `@AfterMethod`.  just initialize this when it's declared on line 80